### DISCUSSION TITLE
Connecting to snowflake in databricks with sparklyr

### DISCUSSION THREAD

**Discussion by usrbinr** [rstudio/sparklyr#3338] [General]
Is there an equivalent way to use sparklyr to connect to a databricks instance replicating the below sparkR query?

otc_sparkr=SparkR::read.df(    
  source = "snowflake", 
  sfUrl = "xxx.xxx.xxx.com",
  sfUser =  dbutils.secrets.get(scope = "kv-cashflow-sandbox", key = "snowflake-pipeline-user"),
  sfPassword   = dbutils.secrets.get(scope = "kv-cashflow-sandbox", key = "snowflake-pipeline-password"),
  Database = 'db_name',
  Schema = 'schema.name' ,
  Warehouse = 'wh.name',
  query= "SELECT * FROM database.schema.name" ##query specification
)

**Comment by erictleung** [comment]
A few years late, but in case this is helpful to you or anyone else, I found this GitHub gist that was helpful. Here is a template that can be changed to your use case and Snowflake configurations. You will need to change the scope information in the dbutils calls.

```r
library(sparklyr)

SparkR::sparkR.session()
sc <- spark_connect(method="databricks")

df_tbl <- spark_read_source(
  sc=sc, 
  name = "adult",
  source = "snowflake", 
  options = list(
  sfUrl = "ui14160.snowflakecomputing.com:443", # replace this with your own
  sfUser = dbutils.secrets.get(scope = "kv-cashflow-sandbox", key = "snowflake-pipeline-user"),
  sfPassword =  dbutils.secrets.get(scope = "kv-cashflow-sandbox", key = "snowflake-pipeline-password"),
  sfDatabase = "DEMO",
  sfSchema = "databricks_demo",
  sfWarehouse = "DEMO_CLUSTER",
  dbtable = "adult"
  )
)
```

Source: https://gist.github.com/sllynn/a2474fb4dfd2bde234c3dda6d07d1229
